File: LeetCode/Medium/0826-most-profit-assigning-work/0826-most-profit-assigning-work.py
class Solution:
    def maxProfitAssignment(self, difficulty: List[int], profit: List[int], worker: List[int]) -> int:
        n = len(difficulty)
        jobs = list(zip(difficulty, profit))
        jobs.sort()
        worker.sort()
        res = 0
        i = 0
        best = 0
        for w in worker:
            while i < n and jobs[i][0] <= w:
                best = max(best, jobs[i][1])
                i += 1
            res += best
        return res

File: LeetCode/Medium/0826-most-profit-assigning-work/0826-most-profit-assigning-work.cpp
class Solution {
public:
    int maxProfitAssignment(vector<int>& difficulty, vector<int>& profit, vector<int>& worker) {
        int n = difficulty.size();
        vector<pair<int, int>> jobs(n);
        for (int i = 0; i < n; ++i) 
            jobs[i] = {difficulty[i], profit[i]};
        sort(jobs.begin(), jobs.end());
        sort(worker.begin(), worker.end());
        int res = 0, i = 0, best = 0;
        for (int w : worker) {
            while (i < n && w >= jobs[i].first) {
                best = max(best, jobs[i].second);
                i++;
            }
            res += best;
        }
        return res;
    }
};